WordPress/gutenberg
# DataViews Sidebar: Display item count on DataViews sidebar

## What?


Display item count on DataViews sidebar.

## Why?


Closes: https://github.com/WordPress/gutenberg/issues/57072

As per the issue -

In the Pattern management sidebar each category is represented by a menu item. Inside each menu item is a count indicating how many patterns there are in the category.
Therefore, it would be good to also display that on the "Pages" management sidebar, so that anyone can get the count of each view without visiting it.

## How?


Display the page count beside each view.
Acheived by fetching all the pages and then doing a filter using Javascript.

## Testing Instructions





Check the screencast below or the steps mentioned -

- Go to site-editor.
- Go to "Pages".
- On the sidebar, you will see that you have the count of each item/view.

### Testing Instructions for Keyboard


## Screenshots or screencast 

**There is one problem as you can see in the below video, that the "All Pages" count displays three, that's because it just "All the pages" but inside it, it doesn't display the trash pages, because of which it displays only two items. So, should this be the correct behaviour, or the count should also display 2, or the trash items should also be displayed under all pages.**

https://github.com/WordPress/gutenberg/assets/39427067/11c0302f-372f-417b-a79f-c4c26cf4d0e4
